### PR TITLE
Add snippets for `sorry`, comments, docstrings, and declarations

### PR DESCRIPTION
**Pull request by jesse-michael-han**
Add snippet for license boilerplate

Use mirror in license-boilerplate

Add def-namespace and inductive-namespace

Add .gitignore

### CODE REVIEW

File: snippets/lean-mode/license-boilerplate [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# -*- mode: snippet -*-
+# name: license-boilerplate
+# key: license
+# --
+/-
+Copyright (c) `(format-time-string "%Y")` $1. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): $1 $2
+
+$3
```
Comment (cipher1024) [inline snippets/lean-mode/license-boilerplate]: Should we have a default string like `put description here` for `$3`?

File: snippets/lean-mode/have [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# -*- mode: snippet -*-
+# name: have
+# key: have
+# --
+have ${1:this} : $2,
+  by {$3},
```
Comment (cipher1024) [inline snippets/lean-mode/have]: can we put `by` as part of the default value in this and the other snippets?

File: snippets/lean-mode/let [hunk 3]
```diff
@@ -3,4 +3,5 @@
 # key: let
 # --
 let ${1:this} : $2,
-  by {$3},
+  ${3:by {$4}}
```
Comment (cipher1024) [inline snippets/lean-mode/let]: for this one, we could default to `:=` and put it on the previous line

Comment (jesse-michael-han) [reply snippets/lean-mode/let]: done; I also removed the type ascription since it's not common to use it with `let`

**Pull request merged**